src/themes/themesController.ts

```diff
@@ -0,0 +1,461 @@
+        }
+        if (!theme["secondary-button-focus-border-color"]) {
+            let secondaryBtnBorderElement: string[] = theme["secondary-button-border"].split(" ");
+            currentTheme["secondary-button-focus-border-color"] = secondaryBtnBorderElement[secondaryBtnBorderElement.length - 1];
+        }
+
+    }
+
+    /**
+     * Set missing color from another provided color
+     * 
+     * e.g.: 
+     * provider: #ffffff; missing: rgba(255, 255, 255, transparencyFactor)
+     * provider: rgb(255, 255, 255); missing: rgba(255, 255, 255, transparencyFactor)
+     * provider: rgb(255, 255, 255, 1); missing: rgba(255, 255, 255, transparencyFactor)
+     * 
+     * @param {keyof ITheme} missing missing property in the theme
+     * @param {string} provider the provided property in the theme
+     * @param {number} transparencyFactor the alpha channel that will be used in the missing property
+     * @param {ITheme} currentTheme the target theme that will be set
+     */
+    private static setMissingColorFromAnotherProperty(missing: keyof ITheme,
+                                               provider: string,
+                                               transparencyFactor: number,
+                                               currentTheme: ITheme): void {
+
+        // provider: #ffffff; missing: rgba(255, 255, 255, transparencyFactor)
+        if (provider.startsWith("#")) {
+            let hexColor = parseInt(provider.substring(1), 16);
+            let r = (hexColor >> 16) & 255;
+            let g = (hexColor >> 8) & 255;
+            let b = hexColor & 255;
+
+            currentTheme[missing] = "rgba(" + r + ", " + g + ", " + b + ", " + transparencyFactor + ")";
+        }
+        // provider: rgb(255, 255, 255); missing: rgba(255, 255, 255, transparencyFactor)
+        else if (provider.startsWith("rgb(")) {
+            let missingColor = "rgba" + provider.substring(3, provider.length - 1) + ", " + transparencyFactor + ")";
+            currentTheme[missing] = missingColor;
+        }
+        // provider: rgb(255, 255, 255, 1); missing: rgba(255, 255, 255, transparencyFactor)
+        else if (provider.startsWith("rgba")) {
+            let rgbIdx = provider.lastIndexOf(",");
+            let transparency = provider.substring(rgbIdx + 1, provider.length - 1);
+            
+            let newTransparency = parseFloat(transparency.trim()) * transparencyFactor;
+            let missingColor = provider.substring(0, rgbIdx + 1) + newTransparency.toFixed(2) + ")";
+            currentTheme[missing] = missingColor;
+        }
+    }
+
+    /**
+     * Create theme elements for applying the theme
+     */
+    public static createThemeStyle(): void {
+
+        // Create banner styles
+        let bannerStyle = document.createElement('style');
+        bannerStyle.type = 'text/css';
+        bannerStyle.id = 'bannerStyle';
+        document.head.appendChild(bannerStyle);
```

Comment:
We are creating too many `style` elements here. Around 20 just for our banner. That creates a lot of noise in the original web site and the web site owners wouldn't be happy about it.

I suggest we have only one style element, and put all the styles into it.